### PR TITLE
[MIG][9.0] medical_base_us

### PR DESCRIPTION
**Pull request by woodbrettm**
Hi there,

This module sets the stage for Luhn Formula Validations on U.S. Social Security Numbers and NPI's, and validations on DEA ID's for other modules to inherit from and use for implementing validation features.

### CODE REVIEW

**Review by lasley — APPROVED** [review]
Thanks @bmw95 

@t3ddftw please review

File: medical_base_us/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Medical Base - US",
+    "summary": "Provides some helper mixins for US locale.",
+    "version": "9.0.1.0.0",
+    "category": "Medical",
+    "website": "https://laslabs.com",
+    "author": "LasLabs, Odoo Community Association (OCA)",
+    "licence": "AGPL-3",
```
Comment (tedsalmon) [inline medical_base_us/__openerp__.py]: s/licence/license

**Review by tedsalmon — APPROVED** [review]
LGTM minus a lint issue in the `__openerp__.py` file. 👍

File: medical_base_us/models/medical_abstract_dea.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            col_name: ``str`` Name of db column to constrain
+            country_col: ``str`` Name of db country column to verify
+        Raises:
+            ValidationError: If constrain is a failure
+            AttributeError: If country column is not valid or is null in db
+        """
+
+        for rec_id in self:
+            if getattr(rec_id, country_col).code == 'US':
+                if self._dea_is_valid(rec_id[col_name]):
```
Comment (susport) [inline medical_base_us/models/medical_abstract_dea.py:78]: Why here not rec_id._dea_is_valid(....) in @api.multi

File: medical_base_us/models/medical_abstract_npi.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
```
Comment (susport) [inline medical_base_us/models/medical_abstract_npi.py]: Order by name the import,  reference in OCA guide

File: medical_base_us/models/medical_abstract_dea.py (hunk 2 above)
Comment (lasley) [reply medical_base_us/models/medical_abstract_dea.py:78]: While I agree that the use of multi would be cleaner in this context, using `model` was an architectural decision to allow for on the fly validation of these numbers without requiring a recordset. 

This is required to allow for validations on form fields for website & some other planned features. Interestingly enough, this played well into the [v10 upgrades](https://github.com/laslabs/vertical-medical/blob/release/10.0/medical_base_us/models/medical_abstract_dea.py) which utilizes `partner-contact/partner_identification` to get rid of the constrain helpers in all of these models. That module required a model method too, so it kind of worked out.

Comment (susport) [reply medical_base_us/models/medical_abstract_dea.py:78]: 👍 Thank

**Comment by woodbrettm** [comment]
@susport thanks for catching the import ordering and @t3ddftw the license spelling :) @lasley updated and ready!

**Review by lasley — APPROVED** [review]
Thanks @bmw95

**Pull request merged**